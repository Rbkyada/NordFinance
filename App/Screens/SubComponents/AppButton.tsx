import React, { memo } from 'react';
import {
  StyleSheet,
  Pressable,
  ActivityIndicator,
  StyleProp,
  ViewStyle,
} from 'react-native';
import LightTheme from '@Theme/LightTheme';
import CommonStyle from '@Theme/CommonStyle';
import { CustomText } from '@CommonComponent/index';

const styles = StyleSheet.create({
  gradientBtn: {
    height: 56,
    paddingHorizontal: 25,
    minWidth: 100,
    borderWidth: 1,
    ...CommonStyle.center,
  },
  alignSelf: {
    alignSelf: 'center',
  },
  marginVertical: { marginVertical: 5 },
});

interface GradientButtonProps {
  title: string | JSX.Element;
  onPress: () => void;
  style?: StyleProp<ViewStyle>;
  isProcessing?: boolean;
  textOnly?: boolean;
  textColor?: string;
  outerStyle?: StyleProp<ViewStyle>;
}
const ButtonComponent = memo((props: GradientButtonProps) => {
  const {
    title,
    onPress,
    style,
    isProcessing = false,
    textOnly = false,
    textColor,
    outerStyle,
  } = props;

  const { marginVertical } = styles;
  return (
    <Pressable
      onPress={() => onPress()}
      disabled={isProcessing}
      style={[marginVertical, outerStyle, style && style]}
      android_ripple={CommonStyle.androidRipple}>
      {((!isProcessing || textOnly) && (
        <CustomText xlarge style={[{ color: textColor || LightTheme.tint }]}>
          {title}
        </CustomText>
      )) || <ActivityIndicator color={LightTheme.tint} />}
    </Pressable>
  );
});

export { ButtonComponent };
