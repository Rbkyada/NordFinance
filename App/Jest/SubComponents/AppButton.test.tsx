import React from 'react';
import renderer from 'react-test-renderer';
import { Pressable } from 'react-native';
import { ButtonComponent } from '@SubComponents/AppButton';

describe('ButtonComponent', () => {
  it('renders correctly with title', () => {
    const title = 'Submit';
    const onPress = jest.fn();
    const tree = renderer
      .create(<ButtonComponent title={title} onPress={onPress} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with processing state', () => {
    const title = 'Submit';
    const onPress = jest.fn();
    const isProcessing = true;
    const tree = renderer
      .create(
        <ButtonComponent
          title={title}
          onPress={onPress}
          isProcessing={isProcessing}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls onPress function when pressed', () => {
    const title = 'Submit';
    const onPress = jest.fn();
    const component = renderer.create(
      <ButtonComponent title={title} onPress={onPress} />,
    );
    const button = component.root.findByType(Pressable);
    button.props.onPress();
    expect(onPress).toHaveBeenCalled();
  });

  it('renders correctly with text-only mode', () => {
    const title = 'Submit';
    const onPress = jest.fn();
    const textOnly = true;
    const tree = renderer
      .create(
        <ButtonComponent title={title} onPress={onPress} textOnly={textOnly} />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls onPress function when pressed', () => {
    const title = 'Submit';
    const onPress = jest.fn();
    const component = renderer.create(
      <ButtonComponent title={title} onPress={onPress} />,
    );
    const touchable = component.root.findByType(Pressable);
    touchable.props.onPress();
    expect(onPress).toHaveBeenCalled();
  });
});
