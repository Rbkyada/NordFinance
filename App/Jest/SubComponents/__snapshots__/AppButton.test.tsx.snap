// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent renders correctly with processing state 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "marginVertical": 5,
      },
      undefined,
      undefined,
    ]
  }
>
  <ActivityIndicator
    color="#FFF"
  />
</View>
`;

exports[`ButtonComponent renders correctly with text-only mode 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "marginVertical": 5,
      },
      undefined,
      undefined,
    ]
  }
>
  <Text
    numberOfLines={0}
    style={
      [
        {
          "fontSize": 18,
        },
        {
          "color": "#1c1c1e",
        },
        {
          "fontFamily": "DMSans-Regular",
        },
        [
          {
            "color": "#FFF",
          },
        ],
      ]
    }
    xlarge={true}
  >
    Submit
  </Text>
</View>
`;

exports[`ButtonComponent renders correctly with title 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "marginVertical": 5,
      },
      undefined,
      undefined,
    ]
  }
>
  <Text
    numberOfLines={0}
    style={
      [
        {
          "fontSize": 18,
        },
        {
          "color": "#1c1c1e",
        },
        {
          "fontFamily": "DMSans-Regular",
        },
        [
          {
            "color": "#FFF",
          },
        ],
      ]
    }
    xlarge={true}
  >
    Submit
  </Text>
</View>
`;
